### PR TITLE
Lenovo.QuickClean version 1.02.0051

### PR DESCRIPTION
**Pull request by philjorgensen**
- [ ] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [ ] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`? 
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [ ] Does your manifest conform to the [1.1 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.1.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/67499)

### CODE REVIEW

**Review by Trenly — DISMISSED** [review]

**Comment by yao-msft** [comment]
This program can only be installed on versions of Windows designed for the following processor architectures:

x64

when running x86 installer on x64 machine.

This manifest should probably utilize UnsupportedOSArchitectures manifest entries.

**Comment by philjorgensen** [comment]
@Trenly 
I supplied the silent install switches and tested the manifest locally, which went through unattended. What am I missing with this one?

**Comment by philjorgensen** [comment]
> This program can only be installed on versions of Windows designed for the following processor architectures:
> 
> x64
> 
> when running x86 installer on x86 machine.
> 
> This manifest should probably utilize UnsupportedOSArchitectures manifest entries.

This shouldn't be the case as I've submitted other Lenovo packages that are x86 installers. I tested the manifest locally and there was no issue.

**Comment by yao-msft** [comment]
This is the full log:
﻿2022-07-25 16:25:10.906 Log opened. (Time zone: UTC-07:00)
2022-07-25 16:25:10.906 Setup version: Inno Setup version 5.5.9 (u)
2022-07-25 16:25:10.906 Original Setup EXE: C:\Users\VALIDA~1\AppData\Local\Temp\WinGet\Lenovo.QuickClean.1.02.0051\lenovo_quick_clean_1.02.0051.exe
2022-07-25 16:25:10.906 Setup command line: /SL5="$2006A,443471,198656,C:\Users\VALIDA~1\AppData\Local\Temp\WinGet\Lenovo.QuickClean.1.02.0051\lenovo_quick_clean_1.02.0051.exe" /SP- /SILENT /NORESTART /LOG="C:\Users\Validator\AppData\Local\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir\WinGet-Lenovo.QuickClean.1.02.0051-2022-07-25-16-25-04.664.log"
2022-07-25 16:25:10.922 Windows version: 10.0.19042 (NT platform: Yes)
2022-07-25 16:25:10.922 64-bit Windows: No
2022-07-25 16:25:10.922 Processor architecture: x86
2022-07-25 16:25:10.922 User privileges: Administrative
2022-07-25 16:25:10.922 64-bit install mode: No
2022-07-25 16:25:10.922 Message box (OK):
This program can only be installed on versions of Windows designed for the following processor architectures:

x64

**Comment by philjorgensen** [comment]
> This is the full log: ﻿2022-07-25 16:25:10.906 Log opened. (Time zone: UTC-07:00) 2022-07-25 16:25:10.906 Setup version: Inno Setup version 5.5.9 (u) 2022-07-25 16:25:10.906 Original Setup EXE: C:\Users\VALIDA~1\AppData\Local\Temp\WinGet\Lenovo.QuickClean.1.02.0051\lenovo_quick_clean_1.02.0051.exe 2022-07-25 16:25:10.906 Setup command line: /SL5="$2006A,443471,198656,C:\Users\VALIDA~1\AppData\Local\Temp\WinGet\Lenovo.QuickClean.1.02.0051\lenovo_quick_clean_1.02.0051.exe" /SP- /SILENT /NORESTART /LOG="C:\Users\Validator\AppData\Local\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir\WinGet-Lenovo.QuickClean.1.02.0051-2022-07-25-16-25-04.664.log" 2022-07-25 16:25:10.922 Windows version: 10.0.19042 (NT platform: Yes) 2022-07-25 16:25:10.922 64-bit Windows: No 2022-07-25 16:25:10.922 Processor architecture: x86 2022-07-25 16:25:10.922 User privileges: Administrative 2022-07-25 16:25:10.922 64-bit install mode: No 2022-07-25 16:25:10.922 Message box (OK): This program can only be installed on versions of Windows designed for the following processor architectures:
> 
> x64

Interesting. Testing this using .\Tools\SandboxTest.ps1 -Manifest <path to quick clean manifest> resulted in a successful install with no prompts. Where did you obtain the above log from?

**Comment by yao-msft** [comment]
Let's give it a retry and see.

**Comment by yao-msft** [comment]
Still same error
image

**Comment by philjorgensen** [comment]
> Still same error image

Any idea why this works with the SandboxTest.ps1 script?

**Comment by yao-msft** [comment]
> Any idea why this works with the SandboxTest.ps1 script?

Looks like sandbox ps1 is created for x64 testing? I saw dependencies only declare x64 vclibs [here](https://github.com/microsoft/winget-pkgs/blob/c9bd3163becb345839f50fde766f169089b79558/Tools/SandboxTest.ps1#L106)

This should be reproducible on a x86 machine.

Thanks.

File: manifests/l/Lenovo/QuickClean/1.02.0051/Lenovo.QuickClean.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Created using wingetcreate 1.0.4.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.1.0.schema.json
+
+PackageIdentifier: Lenovo.QuickClean
+PackageVersion: 1.02.0051
+Installers:
+- MinimumOSVersion: 10.0.0.0
+  Architecture: x86
```
Comment (ItzLevvie) [inline manifests/l/Lenovo/QuickClean/1.02.0051/Lenovo.QuickClean.installer.yaml:8]:
```suggestion
  Architecture: x64
```

**Review by ItzLevvie — CHANGES_REQUESTED** [review]

File: manifests/l/Lenovo/QuickClean/1.02.0051/Lenovo.QuickClean.locale.en-US.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+PackageIdentifier: Lenovo.QuickClean
+PackageVersion: 1.02.0051
+PackageLocale: en-US
+Publisher: Lenovo
+PublisherUrl: https://www.lenovo.com
+PublisherSupportUrl: https://support.lenovo.com
+PrivacyUrl: https://www.lenovo.com/privacy
+Author: Lenovo
+PackageName: Quick Clean
```
Comment (ItzLevvie) [inline manifests/l/Lenovo/QuickClean/1.02.0051/Lenovo.QuickClean.locale.en-US.yaml:12]:
```suggestion
PackageName: Lenovo Quick Clean
```

**Review by ItzLevvie — CHANGES_REQUESTED** [review]

**Review by ItzLevvie — APPROVED** [review]

**Pull request merged**